directus-solidstart/index.md

```diff
@@ -0,0 +1,314 @@
+
+It may be useful to create an `.env` file and store the URL there in case you use different projects for production, development, etc. Otherwise it is not a private value, access is managed through the Directus dashboard.
+
+## Using Global Metadata and Settings
+
+In your Directus project, navigate to Settings -> Data Model and create a new collection called **global**. Under the Singleton option, select 'Treat as a single object', as this collection will have just a single entry containing global website metadata.
+
+![A form named "Global" has two inputs - a title and a description, each filled with some text.](solid-global-config.webp)
+
+By default, new collections are not accessible to the public. Navigate to Settings -> Roles & Permissions -> Public and give Read access to the Global collection.
+
+## Fetch Global Data
+
+Now that the table is created and we have data added, let's create our first fetcher. In `/lib/directus.js` we will add a new method:
+
+```js
+export const getGlobals = cache(async () => {
+  "use server";
+  try {
+    return await directus.request.readItems("global");
+  } catch (e) {
+    console.error(e);
+
+    return null;
+  }
+}, `globals`);
+```
+
+Notice the `"use server"` directive. This indicates the bundler this method will never run (and therefore will not exist) in the client-side build. The `cache()` method wrapper defines a cache-key ("globals") where this response will be stored. We will try the request to Directus, if the request is not successful we can throw a `404`.
+
+With our fetcher, we can head over to our `src/routes/index.jsx` and create our first loader request.
+
+```jsx
+export const route = {
+  load: () => getGlobals(),
+};
+
+export default function Home() {
+  const pageInfo = createAsync(getGlobals);
+
+  createEffect(() => {
+    if (pageInfo() === null) {
+      throw new Error("404");
+    }
+  });
+
+  return (
+    <main>
+      <Show when={pageInfo()}>{(info) => <h1>{info().title}</h1>}</Show>
+
+      <Show when={pageInfo()}>
+        {(info) => <strong>{info().description}</strong>}
+      </Show>
+    </main>
+  );
+}
+```
+
+The `route` object defines a `load` method that will be called **before** our route is accessed. Then `createAsync` will receive the deferred data as a signal and allow our UI to update. The `<Show>` component is important to make sure we only render our elements once the data has finally arrived at the browser.
+
+## Dynamic Routes With Directus
+
+Create a new collection called pages - make the Primary ID Field a "Manually Entered String" called slug, which will correlate with the URL for the page. For example about will later correlate to the page [localhost:3000/about](http://localhost:3000/about).
+
+Create a text input field called title and a WYSIWYG input field called content. In Roles & Permissions, give the Public role read access to the new collection. Create 3 items in the new collection - here's some sample data.
```

Comment:
Note to self - leave this resolved - move sample data into a common space and link it here.